test/odbc/psqlodbc/toby_tinyint_pg_datatype.cpp

```diff
@@ -0,0 +1,854 @@
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_TINYINT));
+  odbcHandler.CloseStmt();
+
+  // Select * From Table to ensure that it exists
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL1_NAME}));
+
+  // Make sure column attributes are correct
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_DISPLAY_SIZE, // Get the length of the column (size of char in columns)
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &length);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, LENGTH_EXPECTED);    //refer to babelfish_extensions/contrib/babelfishpg_common/sql/numerics.sql
+  //                                        the tinyint created based on smallint type since postgresql does not have
+  //                                        tiny int datatype
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                            2,
+                            SQL_DESC_LENGTH, // Get the Bytes of the column 
+                            NULL,
+                            0,
+                            NULL,
+                            (SQLLEN*) &length);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(length, BYTES_EXPECTED);    //Bytes same as length problem above
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_PRECISION, // Get the precision of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &precision); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(precision, PRECISION_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_SCALE, // Get the scale of the column
+                          NULL,
+                          0,
+                          NULL,
+                          (SQLLEN*) &scale); 
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(scale, SCALE_EXPECTED);
+
+  rcode = SQLColAttribute(odbcHandler.GetStatementHandle(),
+                          2,
+                          SQL_DESC_TYPE_NAME, // Get the type name of the column
+                          name,
+                          BUFFER_SIZE,
+                          NULL,
+                          NULL); 
+  ASSERT_EQ(string(name), NAME_EXPECTED);  //Same as problem above
+
+  rcode = SQLFetch(odbcHandler.GetStatementHandle());
+  ASSERT_EQ(rcode, SQL_NO_DATA);
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_Tinyint, Insertion_Success) {
+
+  const int BYTES_EXPECTED = 1;
+
+  unsigned char pk;
+  unsigned char data;
```

Comment:
Reason for using unsigned char? I saw that you used `int` in one of the tests below, `Table_Composite_Keys()`